packages/rollup-core/src/app/data/consumers/canonical-chain-batch-submitter.ts

```diff
@@ -197,4 +217,164 @@ export class CanonicalChainBatchSubmitter extends ScheduledTask {
     }
     return true
   }
+
+  protected async getBatchSubmissionBlockNumber(): Promise<number> {
+    // TODO: This is bad. Make this not suck.
```

Comment:
Could be nice to change the comment to say what this should be set to eventually. I am not 100% sure I even know what the ideal value would be here

Reply:
Oh and just to make sure I understand, we don't need to do this with timestamp because Geth will handle it's own timestamp updating? Or could it make sense to also have a function `getBatchSubmissionTimestamp()`?

Reply:
Yeah, the timestamp is included in the block output, so it's on the tx objects

Reply:
And yes, I assume block number will have to be handled the same way as timestamp. It'll have to be passed through, and geth will have to use _that_ when handling block number queries.

Reply:
Updated comment to indicate that this should be handled within geth like timestamp 👍